KY-032_AvoidanceTracking/KY-032_AvoidanceTracking.ino

```diff
@@ -1,25 +1,99 @@
+ ** When there is NOTHING detected the digital output == HIGH == TRUE == 1 
+ ** When the sensor detects an obstacle the digital output == LOW == FALSE == 0
+ *
+ * The KY-032 has one jumper (EN) and 4 pins: GND, +, out, EN.
+ *
+ * The KY-032 sensor can be enabled or disabled via hardware or software
+ * With the EN jumper in place, the sensor will be in constant detection mode.
+ *
+ * If you remove the EN jumper you can enable and disable the sensor by sending
+ * a HIGH (enable) or LOW (disable) signal to the EN pin
+ * a HIGH signal will enable the sensor and a LOW signal will disable it.
+ * 
+ *
+ * Library requirements:
+ * None
+ *
+ * Code contributions:
+ *    David Schmidt (davids@inventr.io)
+ *    Frank Visco (bittoe@yahoo.com)
+ *
+.-------------------------------.
+> Avoidance tracker Calibration <
+:---------------------------------------------------------------------------------------:
+| To get this to work you have to adjust both the output and input potentiometers (POT).
+|
+| The POT on the right, which sits just below the EN jumper, adjusts the LED output.
+| -Turn this clockwise (to the right) as far as it will go. 
+|
+| The POT on the left adjusts sensor sensitivity.
+| -Turn this as far counterclockwise (to the left) as it will go.
+| -Place your hand or some other flat object about 4″ (~10cm) above the sensor.
+| -SLOWLY turn the POT clockwise (right) until the LED on the PCB comes on and the serial output says “obstacle in the way”
+|
+| There is a small range for detection but With a little finesse, gentle nudging, and tweaking
+| of the POT you should be able to get a detection range between 0 and ~10″ (~25cm).
+| You really have to play with the detection sensitivity to get it to work reliably and consistently
+:---------------------------------------------------------------------------------------:
+ *
+ * This project uses: 
+ * One Digital pin to INPUT (object detection).
+ * One *optional* digital pin to OUTPUT (enable/disable sensor)
+ * On the Hero (Arduino Uno compatible) we *could* use: D0-D13, A0-A5.
+ * Skip: A0-A5 (save for Analog),
+ *       D0/D1 (used by USB),
+ *       D2/D3 (save for interrupts),
+ *       D13 (used by LED_BUILTIN and SPI Clock),
+ *       D5, D6, D9, D10 and D11 (save for PWM)
+ *       D11 (SPI MOSI)
+ *       D12 (SPI MISO)
+ * Recommended for fewest conflicts:
+ *    D4, D7 or D8
  */
+const uint8_t KY_032_PIN_OUT = 8;   // set sensor output to a digitial pin 
+// If you want the ability to enable and disable the sensor manually 
+// you must remove the EN jumper and 
+// Assign the EN pin to a digital pin:
+//    const uint8_t KY_032_PIN_EN = ; 
+// Set the pin mode
+//    pinMode(KY-032_PIN_EN, OUTPUT);
+// Read the value of the OUTPUT pin
+//    int KY_032_enable = digitalRead(KY_032_PIN_EN);
 
 void setup() {
-  Serial.begin(9600);
-  pinMode(8, INPUT);  // set pin as input
+Serial.begin(9600);
+pinMode(KY_032_PIN_OUT, INPUT);// set pin as input
```

Comment:
These two lines aren't indented.  Quick fix, in the Edit menu look for "Auto Format".  Use that menu or the shortcut key it lists (Cmd-T on the Mac).

Reply:
oops